jina/orchestrate/flow/base.py

```diff
@@ -1512,6 +1520,26 @@ def host(self, value: str):
             self.enter_context(self[GATEWAY_NAME])
             self[GATEWAY_NAME].wait_start_success()
 
+    @property
+    def monitoring(self) -> bool:
+        """Return if the monitoring is enabled
+        .. # noqa: DAR201
+        """
+        if GATEWAY_NAME in self._deployment_nodes:
+            return self[GATEWAY_NAME].args.monitoring
+        else:
+            return False
+
+    @property
+    def port_monitoring(self) -> int:
+        """Return if the monitoring is enabled
+        .. # noqa: DAR201
+        """
+        if GATEWAY_NAME in self._deployment_nodes:
+            return self[GATEWAY_NAME].args.port_monitoring
+        else:
+            return __default_port_monitoring__
```

Comment:
I do not understand this logic. Who is guaranteeing that there is something in this default port?

Reply:
Then what should I return if the gateway_name is not in the deployment node ?

> Who is guaranteeing that there is something in this default port?
Nothing guarantee that, it is just saying on which port the monitoring would be serve if the monitoring is activaed. There is another attribute to say if it is activated or not